docs/adr/0008-use-tracing.md

```diff
@@ -0,0 +1,165 @@
+any "matching" capabilities (eg, regular expressions or similar) - each target will use an exact string match.
+
+When subscribing to a target, the app will also need to specify the max-level for that target. In other words, a
+subscription can be considered a tuple of `(target, level)`.
+
+This implies that checking whether an individual tracing event has a subscriber will be a `HashMap` lookup for the target name,
+then a numeric comparison for the level. Each target will allow only one subscriber, identified by the `target`.
+This means that changing the level for an existing subscriber will involve replacing the subscription with a new one which holds the new level.
+It is expected that the implementation will simply be "set a new subscription", which will replace the old subscription.
+
+This assumes that the application will subscribe with the actual level needed by that subscriber - ie, that apps will not
+simply subscribe to the highest log level and perform additional filtering after the log event has been dispatched.
+This means that the "performance critical path" for this process is determining if we have a subscriber;
+once we have determined a subscriber matches an event, we can perform relatively expensive operations
+on the event because we assume action will be taken on the event.
+
+An example of these "relatively expensive" operations is fetching event "fields", such as the message or other meta-data.
+
+You will note that this is generic enough to handle traditional "log" messages, our error-reporting logs, and more structured logging.
+The shape of the structs etc (eg, try and use one struct for all use-cases, one struct per use-case, etc) is intentionally omitted to avoid bike-shedding on those details in the first instance.
+
+### Replace all existing "subscribers"
+
+There are 3 main places which would change in the first instance, broken down by platform
+
+#### Mobile
+
+The existing [rust_log_forwarder](https://github.com/mozilla/application-services/tree/main/components/support/rust-log-forwarder) and [error-support](https://github.com/mozilla/application-services/tree/main/components/support/error) crates would change how they collect "events" to using `tracing-subscriber`. Initially we'd not change the API exposed to the mobile apps, but very soon after declaring success on Desktop we'd look to consolidate the APIs as seen by the applications.
+
+#### Desktop
+
+Desktop has a [hand-written xpcom-based log adaptor](https://searchfox.org/mozilla-central/source/services/sync/golden_gate/src/log.rs#119-120). This would be removed entirely and a uniffi-based callback mechanism is used. Rust code calling back into Javascript has the same semantics as `golden_gate` - the log calls are "fire and forget", ending up in the main thread automatically.
+
+The [`gecko-logger`](https://searchfox.org/mozilla-central/source/xpcom/rust/gecko_logger/src/lib.rs) crate would change:
```

Comment:
This makes me so happy! 🙌🏼 I felt like the "log sink" pattern started out as a stopgap to connect Rust bookmark sync to Sync's logging machinery, and it was easy for us to generalize once we started adding more Rust components. It's wonderful to revisit it from the ground up 😁